packages/node/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -102,15 +102,15 @@ export class AuthCodeRedirectHandler implements IRedirectHandler {
   }
 
   async handle(
-    redirectUrl: string
+    inputRedirectUrl: string
   ): Promise<ISessionInfo & { fetch: typeof fetch }> {
-    if (!(await this.canHandle(redirectUrl))) {
+    if (!(await this.canHandle(inputRedirectUrl))) {
       throw new Error(
-        `AuthCodeRedirectHandler cannot handle [${redirectUrl}]: it is missing one of [code, state].`
+        `AuthCodeRedirectHandler cannot handle [${inputRedirectUrl}]: it is missing one of [code, state].`
```

Comment:
Hhmmm... interesting code smell here - i.e., this `handle()` method is explaining why a completely different method, `canHandle()`, has returned `false`. Clearly it's not this function's job to be doing that, as once a developer tweaks the `canHandle()` method to add some new 'I can't handle this' condition, and naturally doesn't realize they also need to update this completely different method accordingly, this error message becomes dangerously misleading.
No easy fix here I don't think, but a good example of why the LIT always passed a mutable 'Result' Dataset as an input parameter all along the request processing call stack - i.e., so that each and every processing stage could independently report it's behaviour all within a single 'Result' object (just by using RDF), and why nested errors are so important too (although in this particular case, `canHandle()` is not throwing an error - so the mutable, extensible, self-descriptive 'Result' object would come in here instead).
In this case, I suppose we should add a comment to the `canHandle()` return statement pointing out that the `handle()` method is currently describing this 'missing one of [code, state]' behaviour...?

Reply:
I think the easiest approach would be to change the way the handler pattern is currently implemented to have a separate validation function (currently `canHandle`) that throws when the validation fails, with an error message describing the failure. Then, `canHandle` just has to call this function, and catch the error if a failure happens to still return false (because it's expected that `canHandle` may return false without stopping the app), and `handle` can also call this function, and not catch the error if any. This way, only the validation function implements the validation logic and the attached error message. Since that's something we'd need to apply to all implementations of the handler pattern, I'll add a ticket to this end.